### PR TITLE
IGAPP-825: Language change broken

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

File: native/src/hooks/useClearRouteOnClose.ts [hunk 1]
```diff
@@ -4,17 +4,23 @@ import { Dispatch } from 'redux'
 import { RoutePropType, RoutesType } from '../constants/NavigationTypes'
 import { StoreActionType } from '../redux/StoreActionType'
 
-const useClearRouteOnClose = (route: RoutePropType<RoutesType>, dispatch: Dispatch<StoreActionType>): void => {
+const useClearRouteOnClose = (
+  route: RoutePropType<RoutesType>,
+  dispatch: Dispatch<StoreActionType>,
+  clearRouteOnClose = false
```
Comment (maxammann) [inline native/src/hooks/useClearRouteOnClose.ts:10]: I think this should be a quired property to make clear that this is important. Not clearing by default is weird imo.

**Review by maxammann — APPROVED** [review]

File: native/src/hocs/withPayloadProvider.tsx [hunk 2]
```diff
@@ -66,20 +67,28 @@ export type PropsType<
...
 const withPayloadProvider = <
-  S extends { dispatch: Dispatch<StoreActionType> },
+  S extends {
+    dispatch: Dispatch<StoreActionType>
+  },
   R extends Record<string, unknown>,
   T extends RoutesType
 >(
   refresh: (refreshProps: R, dispatch: Dispatch<StoreActionType>) => void,
+  clearRouteOnClose: boolean,
```
Comment (SteffiStuffel) [inline native/src/hocs/withPayloadProvider.tsx:81]: Why is this needed now?

**Review by SteffiStuffel — APPROVED** [review]
Tested works fine

File: native/src/hocs/withPayloadProvider.tsx (hunk 2 above)
Comment (klinzo) [reply native/src/hocs/withPayloadProvider.tsx:81]: Because not all routes need to clear the route from redux on close. See the changes from #457, before a function `onRouteClose` was passed wherever necessary, but since the function is always the same (just dispatch an action) we don't need to pass it and a boolean is enough.
